warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownLargeFieldedListsVisitor.java

```diff
@@ -178,15 +189,43 @@ public Object visit(ASTOrNode node, Object data) {
             else {
                 // recurse on the eq children in this subset
                 copyChildren(eqNodes, children, data);
+                track(data, field, eqNodes.size() - 1);
                 
                 // recurse on the range children in this subset
                 copyChildren(rangeNodes, children, data);
+                
+                int numBatches = (int) Math.ceil(rangeNodes.size() / (double) Math.max(1, config.getMaxRangesPerRangeIvarator()));
+                numBatches = Math.min(Math.max(1, config.getMaxOrRangeIvarators()), numBatches);
```

Comment:
Instead of copying this code from above, make a method.  That way if we make changes in the future, we only need to change one part of the code.

Alternatively, you could remove these lines, set ShardQueryconfiguration.maxOrRangeIvarators to 1 during your second pushdown visitor pass in the visitor function and just subtract 1 from rangeNodes.size() in the following track call.

Reply:
got it